luigi/contrib/lsf.py

```diff
@@ -0,0 +1,362 @@
+        args += ["-e", "/".join(log_output[0:-1]) + "/job.err"]
+        args += self.extra_bsub_args.split()
+
+        # Find where our file is
+        runner_path = lsf_runner.__file__
+        # assume source is next to compiled
+        if runner_path.endswith("pyc"):
+            runner_path = runner_path[:-3] + "py"
+
+        args += [runner_path]
+        args += [self.tmp_dir]
+
+        # That should do it. Let the world know what we're doing.
+        LOGGER.info(" ".join([str(a) for a in args]))
+
+        # Submit the job
+        run_job_proc = subprocess.Popen(
+            [str(a) for a in args],
+            stdin=subprocess.PIPE, stdout=subprocess.PIPE, cwd=self.tmp_dir)
+        output = run_job_proc.communicate()[0]
+
+        # ASSUMPTION
+        # The result will be of the format
+        # Job <123> is submitted ot queue <myqueue>
+        # So get the number in those first brackets.
+        # I cannot think of a better workaround that leaves logic on the Task side of things.
+        LOGGER.info("### JOB SUBMISSION OUTPUT:" + str(output))
+        self.job_id = int(output.split("<")[1].split(">")[0])
+        LOGGER.info(
+            "Job %ssubmitted as job %s",
+            self.job_name_flag + ' ',
+            self.job_id
+        )
+
+        self._track_job()
+
+        # If we want to save the job temporaries, then do so
+        # We'll move them to be next to the job output
+        if self.save_job_info:
+            LOGGER.info("Saving up temporary bits")
+
+            # dest_dir = self.output().path
+            shutil.move(self.tmp_dir, "/".join(log_output[0:-1]))
+
+        # Now delete the temporaries, if they're there.
+        self._finish()
+
+    def _track_job(self):
+        time0 = 0
+        while True:
+            # Sleep for a little bit
+            time.sleep(self.poll_time)
+
+            # See what the job's up to
+            # ASSUMPTION
+            lsf_status = track_job(self.job_id)
+            if lsf_status == "RUN":
+                self.job_status = RUNNING
+                LOGGER.info("Job is running...")
+                if time0 == 0:
+                    time0 = int(round(time.time()))
+            elif lsf_status == "PEND":
+                self.job_status = PENDING
+                LOGGER.info("Job is pending...")
+            elif lsf_status == "DONE" or lsf_status == "EXIT":
+                # Then the job could either be failed or done.
+                errors = self.fetch_task_failures()
+                if errors == '':
+                    self.job_status = DONE
+                    LOGGER.info("Job is done")
+                    time1 = int(round(time.time()))
+
+                    # Return a near estimate of the run time to with +/- the
+                    # self.poll_time
+                    LOGGER.info(
+                        "### JOB COMPLETED: %s%s in %s seconds",
+                        self.job_name_flag + ' ',
```

Comment:
Why would you concat a space to a interpolated var instead of just including the space in the string to be interpolated?

Reply:
Moved this logic out to pre compose the required string. Adding the space was to avoid having a double space if there was no name specified.